Add July 2025 release blog (#204)
adoptium/adoptium.net: src/asciidoc-pages/docs/aqavit-verification.adoc

= AQAvit Verification
:page-authors: gdams, smlambert, llxia
Version 0.2, 02.03.2022

> The AQAvit project was created to “make quality certain to happen.” AQAvit verification is achieved through the process of running and passing a prescribed and versioned set of tests in the AQAvit test suite. As per the Eclipse Foundation Quality Verification Software License (QVSL) section 5b, a distributor who wants to certify their distributions then shares a summary of the test results by which verification was achieved.

Reviewer: ```suggestion
The AQAvit project was created to “make quality certain to happen.” AQAvit verification is achieved through the process of running and passing a prescribed and versioned set of tests in the AQAvit test suite. The https://www.eclipse.org/legal/documents/eclipse-foundation-quality-verification-suite-license.php[Eclipse Foundation Quality Verification Software License (QVSL)^] requires those who wish to verify their product to share a summary of the test results by which verification was achieved. This document describes how to run the AQAvit verification tests, check verification has passed, and collect the required test results for publication.
```

AQAvit verification is one of the criteria for listing in the link:/marketplace[Adoptium Marketplace^]. Leveraging the AQAvit test suite to ensure the quality of the binaries listed in the Adoptium Marketplace not only communicates to consumers how serious we are about quality, but also consolidates the good verification practices of the Adoptium Working Group members under a centralized effort. AQAvit aligns its test suite standards with the ever-changing requirements of the user base.

== Overview

> The AQAvit test suite is a large set of tests many contributed to the AQAvit project and some pulled from a variety of open-source projects that serve to verify the quality of OpenJDK binaries.  The suite is suitable for testing Java 8 or higher versions on all link:/supported-platforms[supported platforms^].  To verify that binaries can be listed in the marketplace, vendors will clone a specified release of the aqa-tests github repository, configure their test environment, execute and pass the required test targets against each binary they plan to list in the marketplace and make the results of those test runs available as per the QVSL.

Reviewer: ```suggestion
The AQAvit test suite is a large set of tests, many contributed to the AQAvit project and some pulled from a variety of open-source projects, that serve to verify the quality of OpenJDK binaries.  The suite is suitable for testing Java SE 8 or higher versions on all link:/supported-platforms[supported platforms^].  To verify binaries, vendors clone a specified release of the aqa-tests github repository, configure their test environment, execute and pass the required test targets against each binary they plan to verify and make the results of those test runs available as per the QVSL.
```
Need to ensure this applies to all verified binaries, even if they are not destined for the marketplace.

== AQAvit Test Targets to Run
The tests are divided into different groups and those groups are split into 3 levels.  This logical categorization of the tests provides flexibility and granularity and can be visually represented in a grid.  

image::gridview.png["grid view",350,182]

> For the initial releases of AQAvit, the required set of top-level test targets to run are [sanity.functional, extended.functional, special.functional, sanity.openjdk, extended.openjdk, sanity.system, extended.system, sanity.perf, extended.perf].  In subsequent AQAvit releases, targets will be added to raise quality bar even higher.

Reviewer: ```suggestion
For the current release of AQAvit, the required set of top-level test targets to run are [sanity.functional, extended.functional, special.functional, sanity.openjdk, extended.openjdk, sanity.system, extended.system, sanity.perf, extended.perf].  In subsequent AQAvit releases, targets will be added to raise quality bar even higher.
```

Publish <results>.zip
```
|===
====

.Basic Steps to Run AQAvit Test Targets via Github Workflow
[%collapsible]
====

> The AQAvit project created a Github action that allows for running the AQAvit test suite from workflow files.  The run-aqa action in the link:/https://github.com/adoptium/run-aqa[run-aqa repository^] allows users to pass in custom OpenJDK binaries for verification.  Here is an example workflow file that can run sanity level targets on the 3 supported platforms available as github runners: 

Reviewer: ```suggestion
The AQAvit project created a Github action that allows for running the AQAvit test suite from workflow files.  The `run-aqa` action in the link:/https://github.com/adoptium/run-aqa[run-aqa repository^] allows users to pass in custom OpenJDK binaries for verification.  Here is an example workflow file that can run sanity level targets on the 3 supported platforms available as Github runners: 
```

    strategy:
      fail-fast: false
      matrix:
        os: [ubuntu-latest, macos-latest, windows-2016]
        version: [8, 11, 17]
        impl: [hotspot, openj9]
        buildlist: [openjdk, system, functional, perf]
    steps:
    - uses: actions/checkout@v1
    - uses: AdoptOpenJDK/install-jdk@v1
      with:
        version: ${{ matrix.version }}
        targets: JDK_${{ matrix.version }}
        impl: ${{ matrix.impl }}
        source: 'nightly'
    - name: AQA
      uses: ./
      with: 
        version: ${{ matrix.version }}
        jdksource: 'install-jdk'
        build_list: ${{ matrix.buildlist}}
        target: '_sanity'
```
====

.Basic Steps to Run AQAvit Test Targets via Jenkins
[%collapsible]
====
```
# Set Jenkins job parameters
ADOPTOPENJDK_REPO=https://github.com/adoptium/aqa-tests.git
ADOPTOPENJDK_BRANCH=v0.8.0-release
USE_TESTENV_PROPERTIES=true

# Execute test targets
TARGET=sanity.functional and subsequently [extended.functional|special.functional|sanity.openjdk|extended.openjdk|sanity.system|extended.system|sanity.perf|extended.perf]

# Collect and publish results
Collect *.tap file and metadata file
Archive <results>.zip

Publish <results>.zip
```
====

> The tap files and metadata files contain timestamps and information about the binary under test.  This information is collected from java -version output, release file information and querying some system properties during the test run.  The information should match with the binary listed in the marketplace.

Reviewer: ```suggestion
The `.tap` files and metadata files contain timestamps and information about the binary under test.  This information is collected from `java -version` output, release file information and querying some system properties during the test run.  Where applicable, the information should match with the binary listed in the marketplace.
```
What is the name/extension of the metadata file?
